ToDo/EditScreen.swift

```diff
@@ -0,0 +1,156 @@
+//
+//  EditScreen.swift
+//  Tasks
+//
+//  Created by Maximilian Alexander on 8/27/21.
+//
+
+import Combine
+import DittoSwift
+import SwiftUI
+
+class EditScreenViewModel: ObservableObject {
+    @Published var canDelete: Bool = false
+    @Published var body: String
+    @Published var isCompleted: Bool = false
+    @Published var userId: String
+
+    private let _id: String?
+    private let dittoSync = DittoManager.shared.ditto.sync
+    private let dittoStore = DittoManager.shared.ditto.store
+
+    var isNewToDo: Bool {
+        _id == nil
+    }
+    
+    init(toDo: ToDo?) {
+        self._id = toDo?._id
+        canDelete = toDo != nil
+        body = toDo?.body ?? ""
+        isCompleted = toDo?.isCompleted ?? false
+        userId = toDo?.userId ?? ""
+    }
+
+    func save() async {        
+        if let _id = _id { // updating existing ToDo
+            let query = "UPDATE tasks SET isCompleted = \(isCompleted)"
+            + ", userId = '\(userId)'"
+            + " WHERE _id=='\(_id)'"
+            
+            do {
+                try await dittoStore.execute(query: query)
```

Comment:
So, this is more of a question for myself more than anything else, but I was wondering what the recommended style for these things would be: string interpolation or "aggressive" use of `queryArguments`?

```suggestion
            let query = "UPDATE tasks SET isCompleted = :completed"
            + ", userId = :userId"
            + " WHERE _id==:_id"
            
            do {
                try await dittoStore.execute(query: query, arguments: [
                    "isCompleted": isCompleted,
                    "userId": userId,
                    "_id": _id,
                ])
```

I guess this remark may become especially relevant the moment the values themselves may contain escaping charactes, _e.g._:

```swift
let example = "Daniel's example"
store.execute(query: "SELECT * FROM COLLECTION items WHERE sentence == '\(example)'")
```

Reply:
I went with what first came to mind, but I'm all about discovering best practices as I go. Great question. What is your opinion?
cc @skylerjokiel

Reply:
Probably best to use query arguments to showcase the feature